core/interoperation/src/lib.rs

```diff
@@ -98,30 +96,54 @@ impl Interoperation for InteroperationImpl {
 
 impl InteroperationImpl {
     pub fn new(
-        code_hash_map: HashMap<String, H256>,
+        transaction_hash_map: HashMap<u8, H256>,
         program_map: HashMap<H256, Bytes>,
     ) -> ProtocolResult<Self> {
         init_dispatcher(program_map)?;
-        init_code_hashes_map(code_hash_map);
+        init_ckb_transaction_hashes(transaction_hash_map);
         Ok(InteroperationImpl::default())
     }
+
+    pub async fn init_dispatcher_from_rpc<T: CkbClient>(rpc_client: T, tx_hashes: Vec<H256>) {
+        let rpc_tasks = tx_hashes.into_iter().map(|hash| {
+            let ckb_hash = {
+                let mut bytes = [0u8; 32];
+                bytes.copy_from_slice(hash.as_bytes());
+                bytes.into()
+            };
+            let rpc_task = rpc_client.get_transaction(Default::default(), &ckb_hash);
+            (hash, rpc_task)
+        });
```

Comment:
Call the `get_txs_by_hashes` RPC method here to get all the transactions at once.